packages/theme/src/theme.ts

```diff
@@ -182,6 +185,7 @@ export class Theme extends HTMLElement implements ThemeKindProvider {
 
     protected adoptStyles(): void {
         const styles = this.styles; // No test coverage on Edge
+        if (styles.length < 3) return;
```

Comment:
If this general approach works, does `// ensure all three fragments types are loaded before adopting styles`?

Or should this effect the setting of `this.hasAdoptedStyles`? Maybe `this.hasAdoptedStyles = styles.length > 3;` would allow us to stream themes into the UI, but block after there's something available for all three?

Reply:
Does this need protecting? I'd be tempted to just make `Theme.instances` a `Set<Theme>`, add to the set in connectedCallback, and delete from the set in disconnectedCallback. Then in `registerThemeFragment` just call `adoptStyles` for each instance as you did below. Or is there some problem with calling adopt styles several times?

Reply:
I guess the other approach would be to actually evalulate if the current set of reigstered styles satisfies the theme and only call adopt once it does?

Reply:
`adoptStyles` scales up in processing cost depending on the support the browser has for various APIs. I'd like to skip it until it's ready, but it _could_ just run.

`styles.length === 3` is technically the easiest way to actually evaluate if the current set of registered styles satisfies the theme right now. Unless we want to replay the `reduce` twice on the last instance, it seemed like a good idea to use the single call for both meaning.

Reply:
Then yeah I guess this check here is sufficient.